### ISSUE TITLE
Various accessibility problems

### ISSUE THREAD

**Issue by Texou-** [baedert/corebird#525]
Hi,

Could you think of fixing some major (but simple) accessibility problems in Corebird? Here are 2 examples:
- at each tweet, I read "this tweet contains inappropriate images" before each tweet. I'd like this statement to be disablable/enablable by a setting, so that the screen reader utility could directly speak the tweet and author.
- In pref menu, a lot of widgets have no label matching or their matching is not properly done. Thus, when pressing tab to access to a button/checkbox in pref dialog, the screen reader often doesn't say what it matches to.

`o test the app accessibility, you can use accerciser. It shows you what is sent to at-spi, the accessibility bus. Thus, you'll see that at-spi receive the widget, but not label. The assistive techno doesn't know how to use it.

Tell me if any problem or questions.

Regards,

**Comment by baedert** [comment]
> [...] I'd like this statement to be disablable/enablable by a setting[...]

No. How about properly fixing it instead?

> In pref menu, a lot of widgets have no label matching or their matching is not properly done. Thus, when pressing tab to access to a button/checkbox in pref dialog, the screen reader often doesn't say what it matches to.

That should be fixed in master.

**Comment by Texou-** [comment]
Hi,


Le 28/03/2016 13:51, baedert a écrit :
>     [...] I'd like this statement to be disablable/enablable by a
>     setting[...]
> 
> No. How about properly fixing it instead?

Hmmm... What is this actually? Is it a widget? A text written hardcoded?
I see 2 solutions: either making the message shorter (e.g. "tweet with
inappropriate image", or, likely better, moving it after the name of the
author of the tweet. Thus, it would appear as:
tweet, author, and this label. Thus, the screen !eader would just speak
it if the user wants or waits, but not generally.

> 
>     In pref menu, a lot of widgets have no label matching or their
>     matching is not properly done. Thus, when pressing tab to access to
>     a button/checkbox in pref dialog, the screen reader often doesn't
>     say what it matches to.
> 
> That should be fixed in master.

Excellent. Thanks. I'll check.

Regards,

> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/baedert/corebird/issues/525#issuecomment-202353053>
> 

-- 
Jean-Philippe MENGUAL

HYPRA, progressons ensemble

Tél.: 01 84 73 06 61

Mail: contact@hypra.fr

Site Web: http://hypra.fr

**Comment by baedert** [comment]
I was under the impression that the problem with the inappropriate media message is not its position (which I can't control afaik?), but the fact that it appears for every tweet and not just the ones with inappropriate media *and* if the setting is enabled.

**Comment by Texou-** [comment]
Oh a very new approach of the problem. I am out of control of the sync
between what speaks my screen reader and the actual display. Here's a
screenshot. In it, the 1st message: @AEBNZ; the second @GreggBeratan has
the mention "inapporpriate content". Today they are the only tweet. But
usually, I've 10 tweets like this.

Hope it's helpful.

Regards,


Le 28/03/2016 17:14, baedert a écrit :
> I was under the impression that the problem with the inappropriate media
> message is not its position (which I can't control afaik?), but the fact
> that it appears for every tweet and not just the ones with inappropriate
> media /and/ if the setting is enabled.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/baedert/corebird/issues/525#issuecomment-202432796>
> 

-- 
Jean-Philippe MENGUAL

HYPRA, progressons ensemble

Tél.: 01 84 73 06 61

Mail: contact@hypra.fr

Site Web: http://hypra.fr

**Comment by Texou-** [comment]
Hi,

We've just investigated with Samuel for the problem, and found
something. You can do the same using accerciser. Once installed and run,
you see the widgets tree which is sent in at-spi, the accessibility bus.
In Corebird, this exploration shows that one widget, labelled with "Ce
tweet contient des images marquées comme inappropriées" (cee .po file to
see the English text). The status of this object includes "visible".
It's wrong, as this is not visible on the screen. But given the status,
the screen reader shows it. I guess to make it appear just when
required, Corebird change its size or something like that.

In GTK, a feature enables you to tell that this widget is not visible.
To fix the issue, you should enable it as visible only when required.

Hope it's helpful.

Regards,


Le 30/03/2016 22:28, MENGUAL Jean-Philippe a écrit :
> Oh a very new approach of the problem. I am out of control of the sync
> between what speaks my screen reader and the actual display. Here's a
> screenshot. In it, the 1st message: @AEBNZ; the second @GreggBeratan has
> the mention "inapporpriate content". Today they are the only tweet. But
> usually, I've 10 tweets like this.
> 
> Hope it's helpful.
> 
> Regards,
> 
> 
> Le 28/03/2016 17:14, baedert a écrit :
>> I was under the impression that the problem with the inappropriate media
>> message is not its position (which I can't control afaik?), but the fact
>> that it appears for every tweet and not just the ones with inappropriate
>> media /and/ if the setting is enabled.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/baedert/corebird/issues/525#issuecomment-202432796>
>>
> 

-- 
Jean-Philippe MENGUAL

HYPRA, progressons ensemble

Tél.: 01 84 73 06 61

Mail: contact@hypra.fr

Site Web: http://hypra.fr

**Comment by baedert** [comment]
well it's technically "visible" but not on screen. I've had cases in the past where a widget was actually not visible (not even technically) but the screen reader still considered it visible, which is why I don't think that hiding that label would fix anything.

I've tried my luck in the a11y IRC channel but didn't get an answer as to how I can make a11y only consider the visible child of a `GtkStack` and not all of them.

**Comment by Texou-** [comment]
Hi,

Maybe Samuel, Cc, can help. Otherwise you should ask on accessibi;ity
gnome mailing list. Alejandro Pinero and Joanmarie Diggs will likely
reply, they are accessibility experts for GUI. They will be helpful.
They answer there, less on IRC. They've a Tweeter account as well, don't
know wether they use it (e.g. jdiggs).

Regards,



Le 07/04/2016 09:49, baedert a écrit :
> well it's technically "visible" but not on screen. I've had cases in the
> past where a widget was actually not visible (not even technically) but
> the screen reader still considered it visible, which is why I don't
> think that hiding that label would fix anything.
> 
> I've tried my luck in the a11y IRC channel but didn't get an answer as
> to how I can make a11y only consider the visible child of a |GtkStack|
> and not all of them.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/baedert/corebird/issues/525#issuecomment-206744578>
> 

-- 
Jean-Philippe MENGUAL

HYPRA, progressons ensemble

Tél.: 01 84 73 06 61

Mail: contact@hypra.fr

Site Web: http://hypra.fr

**Comment by Texou-** [comment]
Hello,

> Le 07/04/2016 09:49, baedert a écrit :
> > well it's technically "visible" but not on screen. I've had cases in the
> > past where a widget was actually not visible (not even technically) but
> > the screen reader still considered it visible, which is why I don't
> > think that hiding that label would fix anything.

Well, if the screen reader considers it as visible while it does not
have the visible state, it's a bug in the screen reader that should be
fixed there. That however doesn't mean the application shouldn't try to
do things properly and announce whether widgets are visible are not.

As the time for me to answer shows, I don't really have the time to
investigate myself, but you can probably make up a patch to fix the
visibility attribute and let Jean-Philippe test whether this fixes
things. At least it looks like just the way it's supposed to be fixed,
and if the screen reader doesn't notice that, then the screen reader
maintainers will have to fix their software.

Samuel

**Comment by baedert** [comment]
Any chance you could test changes I make to gtk+ master?

**Comment by baedert** [comment]
The GtkStack issue has been fixed upstream: https://git.gnome.org/browse/gtk+/commit/?id=0b0d4765a00706be78f2485ab0937bc8cbb04031 so a stack should now always only show the visible page.

Closing this, if that commit makes problems, report them to gtk+ upstream.

**Comment by Texou-** [comment]
Hi,

Sorry for delay, fully busy so far. But things will be better soon, so
I'll test. Can you just tell me how to get master and deps to bkild?

Regards,


Le 29/05/2016 à 06:38, baedert a écrit :
> Any chance you could test changes I make to gtk+ master?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/baedert/corebird/issues/525#issuecomment-222342390>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AB-FrkL3jdTxEXeHB-Pdz4UzL9XYrkagks5qGRhagaJpZM4H5nlI>.
> 

-- 
Jean-Philippe MENGUAL

HYPRA, progressons ensemble

Tél.: 01 84 73 06 61

Mail: contact@hypra.fr

Site Web: http://hypra.fr

**Comment by baedert** [comment]
> Can you just tell me how to get master and deps to bkild?

Not really, other than "use jhbuild and look at https://wiki.gnome.org/Newcomers". Like, you will definitely need to at least also compile glib.
